evm/v0.5/src/helpers.ts

```diff
@@ -538,7 +604,31 @@ export function hexToBuf(hexstr: string): Buffer {
   return Buffer.from(stripHexPrefix(hexstr), 'hex')
 }
 
-type Hash = ReturnType<typeof ethers.utils.keccak256>
+type Hash = ReturnType<typeof utils.keccak256>
+
+export async function newServiceAgreement(
+  params: Partial<ServiceAgreement>,
+): Promise<ServiceAgreement> {
+  const agreement: Partial<ServiceAgreement> = {}
+  params = params || {}
+  agreement.payment = params.payment || bigNum('1000000000000000000')
+  agreement.expiration = params.expiration || bigNum(300)
+  agreement.endAt = params.endAt || sixMonthsFromNow()
+  agreement.oracles = params.oracles
+  agreement.requestDigest =
+    params.requestDigest ||
+    '0xbadc0de5badc0de5badc0de5badc0de5badc0de5badc0de5badc0de5badc0de5'
+  agreement.aggregator =
+    params.aggregator || '0x3141592653589793238462643383279502884197'
+  agreement.aggInitiateJobSelector =
+    params.aggInitiateJobSelector || '0x12345678'
+  agreement.aggFulfillSelector = params.aggFulfillSelector || '0x87654321'
+  return agreement as ServiceAgreement
```

Comment:
I would avoid this type cast since it can be dangerous. This is telling the typescript compiler to say that it's a full `ServiceAgreement` type when it may still be partial by accident.

Reply:
I would instead suggest something like this:
```ts
export async function newServiceAgreement(
  params: Partial<ServiceAgreement>,
): Promise<ServiceAgreement> {
  const agreement: ServiceAgreement = {
    payment: bigNum('1000000000000000000'),
    expiration: bigNum(300),
    endAt: sixMonthsFromNow(),
    oracles: [],
    requestDigest:
      '0xbadc0de5badc0de5badc0de5badc0de5badc0de5badc0de5badc0de5badc0de5',
    aggregator: '0x3141592653589793238462643383279502884197',
    aggInitiateJobSelector: '0x12345678',
    aggFulfillSelector: '0x87654321',
    ...params,
  }

  return agreement
}
```